src/integration-tests/workload/deploy_workload_test.go

```diff
@@ -3,60 +3,112 @@ package workload_test
 import (
 	"fmt"
 	"net/http"
+	"regexp"
 	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	"github.com/onsi/gomega/gexec"
 )
 
-var _ = Describe("Deploy workload", func() {
+var ipAddressError = "No IP address found for service"
+
+func getServiceIP() string {
+	timeout := time.After(60 * time.Second)
+	tick := time.Tick(500 * time.Millisecond)
+
+	numBlock := "(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])"
+	validIP := regexp.MustCompile(numBlock + "\\." + numBlock + "\\." + numBlock + "\\." + numBlock)
+
+	for {
+		select {
+		case <-timeout:
+			return ipAddressError
```

Comment:
We should declare `getServiceIP` can return an error in the signature and make `ipAddressError` a real error.